test/utils/testutil/testutil.go

```diff
@@ -16,9 +16,95 @@ limitations under the License.
 
 package testutil
 
-import "os"
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"runtime"
+)
 
 func IsRunningInProw() bool {
 	_, ok := os.LookupEnv("AZURE_CREDENTIALS")
 	return ok
 }
+
+// TestError is used to define the errors given by different kinds of OS
+// Implements the `error` interface
+type TestError struct {
+	LinuxError   error
+	WindowsError error
+	DarwinError  error
+}
+
+// Error returns the error on the basis of the platform
+func (t TestError) Error() string {
+	switch runtime.GOOS {
+	case linux:
+		if t.LinuxError == nil {
+			return ""
+		}
+		return t.LinuxError.Error()
+	case windows:
+		if t.WindowsError == nil {
+			return ""
+		}
+		return t.WindowsError.Error()
+	case darwin:
+		if t.DarwinError == nil {
+			return ""
+		}
+		return t.DarwinError.Error()
+	default:
+		return fmt.Sprintf("could not find error for ARCH: %s", runtime.GOOS)
+	}
+}
+
+const (
+	windows = "windows"
+	linux   = "linux"
+	darwin  = "darwin"
+)
+
+// InitNewTestError returns a TestError{} with the corresponding errors across
+// different platforms
+func InitNewTestError(linuxError, windowsError, darwinError error) TestError {
+	return TestError{
+		LinuxError:   linuxError,
+		WindowsError: windowsError,
+		DarwinError:  darwinError,
```

Comment:
is `DarwinError` necessary? I thought it's same as LinuxError.

Reply:
I noticed in [this](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/555) PR that some test cases have slight variation in the error message across Darwin and Linux, so I thought it might make sense to have a separate field in this case

Reply:
maybe not, let's remove it first. Usually macos test error should be same as linux.

Reply:
and also pls squash commits, thanks.